WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/management/InsightsManagementViewModel.kt

```diff
@@ -2,24 +2,115 @@ package org.wordpress.android.ui.stats.refresh.lists.sections.insights.managemen
+import org.wordpress.android.fluxc.store.StatsStore
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.ALL_TIME_STATS
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.ANNUAL_SITE_STATS
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.COMMENTS
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.FOLLOWERS
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.FOLLOWER_TOTALS
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.LATEST_POST_SUMMARY
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.MOST_POPULAR_DAY_AND_HOUR
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.POSTING_ACTIVITY
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.PUBLICIZE
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.TAGS_AND_CATEGORIES
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.TODAY_STATS
 import org.wordpress.android.modules.UI_THREAD
-import org.wordpress.android.ui.pages.SnackbarMessageHolder
-import org.wordpress.android.util.map
+import org.wordpress.android.ui.stats.refresh.INSIGHTS_USE_CASE
+import org.wordpress.android.ui.stats.refresh.lists.BaseListUseCase
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type.ADDED
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type.REMOVED
+import org.wordpress.android.ui.stats.refresh.utils.StatsSiteProvider
 import org.wordpress.android.viewmodel.ScopedViewModel
+import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 import javax.inject.Named
 
 class InsightsManagementViewModel @Inject constructor(
-    @Named(UI_THREAD) mainDispatcher: CoroutineDispatcher
+    @Named(UI_THREAD) mainDispatcher: CoroutineDispatcher,
+    @Named(INSIGHTS_USE_CASE) val insightsUseCase: BaseListUseCase,
+    private val siteProvider: StatsSiteProvider,
+    private val statsStore: StatsStore
 ) : ScopedViewModel(mainDispatcher) {
-    private val mutableSnackbarMessage = MutableLiveData<Int>()
+    private val _removedInsights = MutableLiveData<List<InsightModel>>()
+    val removedInsights: LiveData<List<InsightModel>> = _removedInsights
 
-    val showSnackbarMessage: LiveData<SnackbarMessageHolder> = mutableSnackbarMessage.map {
-        SnackbarMessageHolder(it)
+    private val _addedInsights = MutableLiveData<List<InsightModel>>()
+    val addedInsights: LiveData<List<InsightModel>> = _addedInsights
+
+    private val _closeInsightsManagement = SingleLiveEvent<Unit>()
+    val closeInsightsManagement: LiveData<Unit> = _closeInsightsManagement
+
+    fun start() {
+        loadInsights()
```

Comment:
I think we should only load insights the first time. Otherwise they get reloaded on screen rotation.